internal/util/validation/validation.go

```diff
@@ -107,6 +113,68 @@ func ValidateBearerToken(token *string, path string) []error {
 	return asErrors(errs)
 }
 
+func ValidateCSRUsages(u *[]string) []error {
+	errs := field.ErrorList{}
+	requiredAllOf := []string{"client auth", "CA:false"}
+	notAllowed := []string{}
```

Comment:
if you made these maps you can reduce lookups to O(1). That should reduce this to one loop and you can just delete() off the map if required.

```
requiredAllOf := map[string]struct{}{
	"clientAuth": {},
	"CA:false":    {},
}

notAllowed := map[string]struct{}{}
```